main/inc/global.inc.php

```diff
@@ -80,6 +80,22 @@
 // Fix bug in IIS that doesn't fill the $_SERVER['REQUEST_URI'].
 api_request_uri();
 
+// Set web proxy environment variables
+foreach ([
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/https/proxy'),
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/http/proxy'),
+             api_get_configuration_sub_value('proxy_settings/curl_setopt_array/CURLOPT_PROXY'),
+         ] as $value) {
+    if ($value && is_string($value)) {
+        foreach (['https_proxy', 'http_proxy', 'HTTPS_PROXY', 'HTTP_PROXY'] as $envVar) {
```

Comment:
I would add this comment here (it is in your bug report but there's no clear reason explained here as to why we use lowercase and uppercase versions of the same):
// libcurl reads environment variable https_proxy: https://curl.haxx.se/libcurl/c/libcurl-env.html
// Guzzle reads environment variable HTTPS_PROXY